ModularTegustation/tegu_items/prosthetics/resurgence_core.dm

```diff
@@ -0,0 +1,44 @@
+/obj/item/organ/cyberimp/chest/resurgence_core
+	name = "Resurgence Clan Augment: Echo Step"
+	desc = "An augment designed by the resurgence clan, which lets the user teleport a short distance a the cost of their sanity."
+	syndicate_implant = TRUE
+	actions_types = list(/datum/action/item_action/organ_action/use/resurgence_core)
+	implant_overlay = "chest_resurgence_core"
+
+/datum/action/item_action/organ_action/use/resurgence_core
+	var/tp_distance = 2
+
+/datum/action/item_action/organ_action/use/resurgence_core/Trigger()
+	if(!IsAvailable())
+		return
+
+	var/turf/T
+	if(owner.dir == 1)
+		T = locate(owner.x, owner.y + tp_distance, owner.z)
+	if(owner.dir == 2)
+		T = locate(owner.x, owner.y - tp_distance, owner.z)
+	if(owner.dir == 4)
+		T = locate(owner.x + tp_distance, owner.y, owner.z)
+	if(owner.dir == 8)
+		T = locate(owner.x - tp_distance, owner.y, owner.z)
```

Comment:
instead of doing this whole thing, please use the `get_teleport_loc()` proc to get the target turf